src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/EventMap.cs

```diff
@@ -76,7 +77,7 @@ private ImmutableArray<Registry<TEventHandler>> GetRegistries<TEventHandler>(str
         private ImmutableArray<Registry<TEventHandler>> GetRegistries_NoLock<TEventHandler>(string eventName)
             where TEventHandler : class
         {
-            _guard.AssertHasLock();
+            Contract.ThrowIfTrue(_guard.CurrentCount > 0);
```

Comment:
❗ I'm concerned about the change in semantics here

Reply:
it's a weaker check.  but will still at least fire when very wrong.